src/RoomsSDKAdapter.js

```diff
@@ -134,4 +144,119 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
 
     return this.getRoomObservables[ID];
   }
+
+  /**
+   * Invoke request to listActivities on adapter
+   *
+   * @param {object} Params to send to conversation api method
+   * @returns
+   * @private
+   */
+  fetchActivities({roomId, earliestActivityDate}) {
+    const {limit} = this;
+    const conversationId = deconstructHydraId(roomId).id;
+
+    logger.debug('ROOM', 'fetchActivities()', ['called with', {conversationId, limit, earliestActivityDate}]);
+
+    let maxDate = earliestActivityDate;
+
+    if (maxDate === null) {
+      maxDate = undefined;
+    }
+
+    return this.datasource.internal.conversation.listActivities({
+      conversationId,
+      limit,
+      lastActivityFirst: true,
+      maxDate,
+    });
+  }
+
+  /**
+   * Handles checking if there is more activities to fetch.
+   *
+   * @param {string} roomId The room id to check
+   * @returns {boolean} True if there is more activities
+   */
+  hasMoreActivities(roomId) {
+    const {earliestActivityDate, activityIds} = this.roomActivities.get(roomId);
+
+    if (activityIds[earliestActivityDate]) {
+      return activityIds[earliestActivityDate] === this.limit;
+    }
+
+    return true;
+  }
+
+  /**
+   * Fetches past activities and returns array of (id, published) objects.
+   *
+   * @param {string} roomId The id of the room
+   * @param {number} limit The max number of activities to return
+   * @returns {external:Observable.<Activity>} Observable stream that emits activity data
+   */
+  async fetchPastActivities(roomId) {
+    const roomActivity = this.roomActivities.get(roomId);
+    let {earliestActivityDate} = roomActivity;
+    const roomObservable = this.roomObservables[roomId];
+
+    if (!this.hasMoreActivities(roomId)) {
+      return roomObservable.complete(roomActivity);
+    }
+
+    try {
+      const data = await this.fetchActivities({
+        roomId,
+        earliestActivityDate,
+      });
+
+      if (!data) {
+        return roomObservable.complete(roomActivity);
+      }
+
+      const activityIds = data.map((activity) => {
+        const {id} = activity;
+
+        roomActivity.activities.set(id, activity);
+
+        return id;
+      });
+
+      const [{published}] = data;
+
+      earliestActivityDate = published;
+      roomActivity.earliestActivityDate = published;
+      roomActivity.activityIds[earliestActivityDate] = activityIds.length;
+
+      this.roomActivities.set(roomId, roomActivity);
+
+      return roomObservable.next(activityIds);
+    } catch (err) {
+      return roomObservable.error(err);
+    }
+  }
+
+  /**
+   * Handle subscribing to previous activities fetched.
+   *
+   * @param {string} roomId The id of the room
+   * @param {number} count The maximum number of activities to return
+   * @returns {external:Observable.<Room>} Observable stream that emits activity data
+   */
+  getPreviousActivities(roomId, count = 50) {
```

Comment:
Follow naming patterns
```suggestion
  getPreviousActivities(roomID, count = 50) {
```